sdk/tests/system/test_event_recording.py

```diff
@@ -0,0 +1,92 @@
+import httpx
+import pytest_asyncio
+from fastapi import Body, HTTPException
+from typing import Annotated
+
+from eidos_sdk.agent.agent import register_program
+from eidos_sdk.agent.client import Agent
+from eidos_sdk.io.events import AgentStateEvent, StringOutputEvent, StartAgentCallEvent, SuccessEvent
+
+
+class HelloWorld:
+    @register_program()
+    async def idle(self, name: Annotated[str, Body()]):
+        if name.lower() == "hello":
+            raise HTTPException(418, "hello is not a name")
+        if name.lower() == "error":
+            raise Exception("big bad server error")
+        return f"Hello, {name}!"
+
+    @register_program()
+    async def idle_streaming(self, name: Annotated[str, Body()]):
+        if name.lower() == "hello":
+            raise HTTPException(418, "hello is not a name")
+        if name.lower() == "error":
+            raise Exception("big bad server error")
+        yield StringOutputEvent(content="Hello, !")
+        yield StringOutputEvent(content=f"{name}!")
+
+
+class TestHelloWorld:
+    @pytest_asyncio.fixture(scope="class")
+    async def server(self, run_app):
+        async with run_app(HelloWorld) as ra:
+            yield ra
+
+    @pytest_asyncio.fixture(scope="function")
+    async def client(self, server):
+        with httpx.Client(base_url=server, timeout=httpx.Timeout(60)) as client:
+            yield client
+
+    def compare_events(self, events, expected_events):
+        assert len(events) == len(expected_events)
+        for event, expected_event in zip(events, expected_events):
+            event_copy = event.copy()
+            expected_event_copy = expected_event.model_dump()
+            del event_copy["create_time"]
```

Comment:
perhaps we want these record flowing out of our original events as well?

Reply:
or perhaps on neither? As a consumer identical streams would be helpful

Reply:
They are two separate things though.  I don't think we want the event time and record id to ever get the to client, IMO.